### PR TITLE
ddl: use callback to refactor column schema test.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
@siddontang  PTAL

### CODE REVIEW

File: ddl/column_test.go [hunk 1]
```diff
@@ -479,67 +492,63 @@ func (s *testIndexSuite) TestAddColumn(c *C) {
...
 	c.Assert(err, IsNil)
 
 	colName := "c4"
 	defaultColValue := int64(4)
+	checkOK := false
 
-	_, err = ctx.GetTxn(true)
-	c.Assert(err, IsNil)
+	tc := &testDDLCallback{}
+	tc.onJobAfterFunc = func(job *model.Job) {
```
Comment (ngaut) [inline ddl/column_test.go]: Replace onJobAfterFunc to onJobDone ?

Comment (siddontang) [inline ddl/column_test.go]: maybe not, because here the job may be not done, but in another new state.

**Comment by siddontang** [comment]
@qiuyesuifeng 
maybe I have a problem when supporting OnJobRunAfter, this callback can not be placed after `d.runJob(t, job)`, because now it is still in the same transaction, so other new transaction can't get the newest changes outer. 

So I think we should put this function after kv.RunInNewTxn.

**Comment by qiuyesuifeng** [comment]
@siddontang  PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**